File: src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/ProtocolTextDocument.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class ProtocolTextDocument(
  val uri: String,
  val filePath: String? = null,
  val content: String? = null,
  val selection: Range? = null,
  val contentChanges: List<ProtocolTextDocumentContentChangeEvent>? = null,
  val visibleRange: Range? = null,
  val testing: TestingParams? = null,
)

File: agent/bindings/csharp/ProtocolTextDocument.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class ProtocolTextDocument
  {
    [JsonProperty(PropertyName = "uri")]
    public string Uri { get; set; }
    [JsonProperty(PropertyName = "filePath")]
    public string FilePath { get; set; }
    [JsonProperty(PropertyName = "content")]
    public string Content { get; set; }
    [JsonProperty(PropertyName = "selection")]
    public Range Selection { get; set; }
    [JsonProperty(PropertyName = "contentChanges")]
    public ProtocolTextDocumentContentChangeEvent[] ContentChanges { get; set; }
    [JsonProperty(PropertyName = "visibleRange")]
    public Range VisibleRange { get; set; }
    [JsonProperty(PropertyName = "testing")]
    public TestingParams Testing { get; set; }
  }
}
